ui/src/pages/destinations/DestinationsPage.tsx

```diff
@@ -138,14 +138,20 @@ export const DestinationsPageContent: React.FC<DestinationsPageContentProps> =
         enqueueSnackbar("Failed to delete components.", { variant: "error" });
       }
     }
+    const queryData = data ?? { destinations: [] };
+      const rows =
+      (queryData as DestinationsQuery).destinations !== undefined
+        ? [...(queryData as DestinationsQuery).destinations]
+        : [...(queryData as DestinationsInConfigsQuery).destinationsInConfigs];
```

Comment:
```suggestion
    const rows = "destinations" in queryData
        ? [...queryData.destinations]
        : [...queryData.destinationsInConfigs];
```

I think this can be a bit cleaner using a type guard, either way `const rows` is indented too far